lumicks/pylake/force_calibration/touchdown.py

```diff
@@ -31,3 +31,36 @@ def fit_piecewise_linear(x, y):
     pars, _ = curve_fit(piecewise_linear, x, y, initial_guess)
 
     return pars
+
+
+def fit_sine_with_polynomial(independent, dependent, freq_guess, freq_bounds, background_degree):
+    """Fit a sine wave plus polynomial background.
+
+    We wish to fit a sine wave (with phase shift) plus a polynomial. By using a trick we can rewrite
+    this equation such that we only have to optimize over 1 variable:
+
+        amp * sin(a * x + phase) + other
+
+    can we written as:
+
+        amp * sin(phase) * cos(a * x) + amp * cos(phase) * sin(a * x) + other
+
+    By absorbing the sin and cosine into the amplitude, we change variables from amplitude and
+    phase, to sine amplitude and cosine amplitude. This means the problem of fitting a sine plus
+    polynomial is essentially a linear one in all variables except the frequency (our parameter of
+    interest). Doing this allows us to estimate this using only one estimated variable."""
```

Comment:
I think adding the parameters to the docstring would be useful here

Reply:
Good point. Not sure why I forgot them. Added!